mindmeld/components/custom_action.py

```diff
@@ -0,0 +1,187 @@
+import logging
+
+import aiohttp
+import requests
+
+from .dialogue import DialogueResponder
+
+
+logger = logging.getLogger(__name__)
+
+
+RESPONSE_FIELDS = ["frame", "directives", "params"]
```

Comment:
Why not include all fields in `DialogueResponder` like `slots`, `dialogue_state` and `history`?

Reply:
Because I don't want the requests to be too large.

Reply:
We should atleast include `slots ` and `dialogue_state `, as discussed in the design meeting where we concluded that these fields are important for slot filling and multi-turn dialogue, two features important for the client requesting this.

Reply:
I see, sure. Make sure that the API doc is reviewed since it is the source of truth.